src/core.js

```diff
@@ -123,6 +123,25 @@ function use(plugin) {
   }
 }
 
+/*
+ * This is what the jsx parser uses. We have to pass in the props,
+ * even though our components have no constructor, to allow for
+ * custom functional components.
+ * From - <Identifier name="_" />
+ * To   - j.createElement(Identifier, { name: '_' }, child1, child2, ...)
+ *
+ * @augments core
+ * @static
+ * @param {}
+ */
+function createElement(Component, props, ...children) {
+  const finalProps = Object.assign(props || {}, { children });
+  const astNode = new Component(finalProps);
+  astNode.__IS_JSX__ = true;
+  astNode.props = finalProps;
+  return astNode;
```

Comment:
Similarly, if you assign `toAST` here, we don't need special treatment in replacement methods (assuming the corresponding changes mentioned there):

```js
matcher.toAST = () => jsxToAST(astNode)
```